### PR TITLE
PlanPrice Component: Update rawPrice conditionals to allow arrays again

### PR DESCRIPTION
**Pull request by jjchrisdiehl**
#### Changes proposed in this Pull Request

Recent [changes to PlanPrice](https://github.com/Automattic/wp-calypso/blob/7f30ae92e681842aac8f812bc7f7192d54c3a2db/client/my-sites/plan-price/index.jsx#L27) added a conditional to check if `rawPrice` is a number, but this broke the functionality for supplying an array of numbers to PlanPrice to create a pricing range:

image

I'm not entirely sure if removing the ability to add a range of prices was on purpose, but it seems we still allow for an array of numbers in the component prop-types so I suspect perhaps it was unintended.

Maybe @mattgawarecki or @monsieur-z could shed some light here as it looks like you did some work on this component some time ago.

There also seems to be some back and forth as to whether we should allow a price of zero to be added as a prop to `PlanPrice`. This PR sets the conditional check to `if ( rawPrice > 0 && rawPriceRange.includes( 0 ) )` to explicitly check if rawPrice is a number greater than zero. 

Updated the dev docs to demonstrate this as well. 

#### Testing instructions

* Open http://calypso.localhost:3000/devdocs/blocks/plan-price while on this branch
* You should see the 'Plan with a price range' working as expected
* You should also see an example using a price of zero

### CODE REVIEW

**Comment by mattgawarecki** [comment]
> I'm not entirely sure if removing the ability to add a range of prices was on purpose, but it seems we still allow for an array of numbers in the component prop-types so I suspect perhaps it was unintended.
>
> Maybe @mattgawarecki or @monsieur-z could shed some light here as it looks like you did some work on this component some time ago.

To be completely honest, I didn't know `PlanPrice` ever supported ranges. Nifty! As far as I can recall, removing this functionality wouldn't have been intentional.

These changes look reasonable enough, and I can see the expected behavior via Calypso Live. The only question I have before adding my own approval is whether we intend to remove decimal points on the upper price in a range, as shown below. Normally I'd expect this to say `2.25 - 3.00`, but what do you think @jjchrisdiehl?

image

**Comment by sirbrillig** [comment]
FWIW Here's where price ranges were implemented: https://github.com/Automattic/wp-calypso/pull/36651 cc @enejb 

> The only question I have before adding my own approval is whether we intend to remove decimal points on the upper price in a range, as shown below. Normally I'd expect this to say 2.25 - 3.00

It looks to me like this applies to all prices, not just the upper price, and that it was intentional to only display decimals if they are non-zero:

https://github.com/Automattic/wp-calypso/blob/cea86a12011e396f86f05c0eafd060c3653e42eb/client/my-sites/plan-price/index.jsx#L51-L54

Personally, my instinct is to not change this behavior here since this is primarily a bug fix PR. @mattgawarecki if you think that prices look better with decimals always included, it might be worth another PR to add that behavior (or maybe something more complex like "always show decimals in a range if any of the prices in the range have a decimal") since I think it would change all price displays.

**Review by mattgawarecki — APPROVED** [review]

File: client/my-sites/plan-price/index.jsx
Comment (sirbrillig) [inline client/my-sites/plan-price/index.jsx]: It occurs to me that `rawPrice > 0` when `rawPrice` is an array will not give us quite the right answer, probably. We might want to replace this with something different.

Comment (sirbrillig) [reply client/my-sites/plan-price/index.jsx]: Maybe this gets closer to the original intention of this guard, to render nothing if a range array is passed that contains a 0:

```suggestion
		if ( Array.isArray( rawPrice ) && rawPriceRange.includes( 0 ) ) {
```

**Comment by enejb** [comment]
Thanks for the ping here. 

As far as I remember this change was done Jetpack Backup had some very specific pricing and different tiers. ( Daily and Instant) See p1HpG7-7DH-p2

This doesn't seem to be the case anymore https://cloud.jetpack.com/pricing 

So I think we can also simplify this component to just use one price.

**Review by sirbrillig — APPROVED** [review]
Looks good! The new tests should make sure that the zero, range, and decimal behaviors continue to work as currently defined for the future unless someone intentionally changes them.

**Comment by jjchrisdiehl** [comment]
Will merge, I'll maintain the current ability to add a range of prices since it could perhaps be used again in the future.

**Pull request merged**
